soc/arm/microchip_mec/mec1501/power.c

```diff
@@ -55,19 +53,13 @@ static void z_power_soc_deep_sleep(void)
 	/*
 	 * Unmask all interrupts in BASEPRI. PRIMASK is used above to
 	 * prevent entering an ISR after unmasking in BASEPRI.
-	 * We clear PRIMASK in exit post ops.
 	 */
-	base_pri = __get_BASEPRI();
 	__set_BASEPRI(0);
```

Comment:
Since you are, now, un-conditionally clearing BASEPRI, without restoring it at the end, I believe it would be nice to document this in the function (i.e. it un-conditionally un-masks interrupts)

Reply:
The absence of restoring BASEPRI is due to the lone `arch_irq_lock()` in `idle()` without a corresponding `arch_irq_unlock()`. That does not seem right. However, since other SoCs are also unmasking interrupts unconditionally, this mimics the same behavior. IMHO, we will need to do something about the lone `arch_irq_lock()` where BASEPRI needs to be restored properly. This will require some fundamental and possibly API changes. Therefore, a workaround is put here first to unblock development, while we can look into fixing idle handling.